tests/test_util.py

```diff
@@ -0,0 +1,20 @@
+from flow.util.misc import bidict
+
+
+class TestBidict:
+    def test_bidict(self):
+        # The bidict class inherits from dict, so we trust its inherited
+        # methods and only need to validate its overrides and inverse behavior.
+        bd = bidict({"a": 1, "b": 2})
+        bd["c"] = 1
+        assert bd == {"a": 1, "c": 1, "b": 2}
+        assert bd.inverse == {1: ["a", "c"], 2: ["b"]}
```

Comment:
I am not sure about this, but will the order always be conserved?

Reply:
The inverse values should always be insertion-ordered. I considered using a `set` instead of a `list` to store the inverse mapping's values (which are keys in the forward mapping), but it turns out that we need the insertion-ordering property so that aggregate stores return groups in the order that they are registered. This is necessary for ensuring that groups/operations are always ordered in a deterministic way. I will add a comment explaining this requirement.